src/core/thread/key_manager.hpp

```diff
@@ -61,6 +62,124 @@ namespace ot {
  * @{
  */
 
+/**
+ * This class represents Security Policy Rotation and Flags.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class SecurityPolicy : public otSecurityPolicy
+{
+public:
+    enum : uint16_t
+    {
+        kMinKeyRotationTime     = 1,   ///< The minimum Key Rotation Time in hours.
+        kDefaultKeyRotationTime = 672, ///< Default Key Rotation Time (in unit of hours).
+    };
+
+    enum : uint8_t
+    {
+        kThread11FlagsLength =
+            OT_THREAD_1_1_SECURITY_POLICY_FLAGS_LENGTH, ///< The Thread 1.1 Security Policy Flags length.
+        kThread12FlagsLength =
+            OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH, ///< The Thread 1.2 Security Policy Flags length.
+    };
+
+private:
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+    enum : uint16_t{
+        kDefaultFlags = 0xfff8, ///< Default Security Policy Flags.
+    };
+#else
+    enum : uint8_t
+    {
+        kDefaultFlags = 0xff, ///< Default Security Policy Flags.
+    };
+#endif
+
+public:
+    /**
+     * This constructor initializes the object with default Key Rotation Time
+     * and Security Policy Flags.
+     *
+     */
+    SecurityPolicy(void)
+    {
+        mRotationTime = kDefaultKeyRotationTime;
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+        Encoding::BigEndian::WriteUint16(kDefaultFlags, mFlags);
+        mFlagsLength = kThread12FlagsLength;
+#else
+        mFlags[0]    = kDefaultFlags;
+        mFlagsLength = kThread11FlagsLength;
+#endif
+    }
```

Comment:
Suggestion: How about adding a method `SetToDefault()` and then use it in constructor?
The reason is to help `mSecurityPolicy  = SecurityPolicy();` use the same method in the `dataset.cpp`.